recipes/server_check.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require_relative './server_check_helper'
+
+consul_service_folder = '/etc/systemd/system/consul.service.d'
+drop_in_file = "#{consul_service_folder}/server_start_prevention.conf" # prevent systemd from starting
+
+directory consul_service_folder do
+  owner 'root'
+  mode '0755'
+  action :create
+  recursive true
+  not_if { ::File.directory?(consul_service_folder) }
+end
+
+# Output file is used for systemd to check whether it should start consul.
+if is_server_allowed?(node['consul']['config']['retry_join'], node['ipaddress'])
```

Comment:
`retry_join` seems to be an already existing attribute, however I could not find any documentation about it in this cookbook. It might be worth updating the README and document what it is, it would make it easier to understand this in the future.

Reply:
Added some information about retry_join in the comments.